network/conduit.go

```diff
@@ -3,57 +3,23 @@
 package network
 
 import (
+	"context"
 	"errors"
 	"fmt"
-	"sort"
 
 	"github.com/onflow/flow-go/model/flow"
 )
 
-// Channel specifies a virtual and isolated communication medium.
-// Nodes subscribed to the same channel can disseminate epidemic messages among
-// each other, i.e.. multicast and publish.
-type Channel string
-type ChannelList []Channel
+// ConduitFactory is an interface type that is utilized by the Network to create conduits for the channels.
+type ConduitFactory interface {
+	// RegisterAdapter sets the Adapter component of the factory.
+	// The Adapter is a wrapper around the Network layer that only exposes the set of methods
+	// that are needed by a conduit.
+	RegisterAdapter(Adapter) error
```

Comment:
so we'll be able to pass different `Adapter`s to the `ConduitFactory` depending on whether it's the real or the testing `Adapter`?

A full diagram showing how the `Adapter`, `ConduitFactory`, `Conduit`, `Channel` and the network fit together would be helpful

Reply:
We resolved this in our meeting.